### PR TITLE
Feature/ed4 ftr 7796 parsing initial file

### PR DESCRIPTION
**Pull request by jpolack**

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -6,10 +6,23 @@ import (
...
+	importInfo, err := postalReturn.Import(postalReturn.ImportConfig{
+		FileInitial: "../pkg/files/rueck.csv",
+		StoreConfig: postalReturn.StoreConfig{
+			User:       "postgres",
+			Password:   "mysecretpassword",
+			Host:       "localhost",
+			Port:       5432,
+			Schema:     "postgres",
+			DisableSSL: true,
+			Verbose:    true,
```
Comment (sfqi) [inline cmd/main.go:18]: Is this hard-coded config here on purpose?

File: pkg/postalReturn/store.go [hunk 2]
```diff
@@ -1 +1,224 @@
...
+
+	return s.db.Save(&storeableImportInformation).Error
+}
+
+func (s *store) CountSkipps() int {
+	var count int
+	err := s.db.Model(&storeableSkipped{}).Count(&count).Error
+
+	if err != nil {
+		log.Fatal(err)
```
Comment (sfqi) [inline pkg/postalReturn/store.go:210]: Maybe return the error instead of fataling?

File: pkg/postalReturn/store.go [hunk 3]
```diff
@@ -1 +1,224 @@
...
+
+	return count
+}
+
+func (s *store) CountProcesses() int {
+	var count int
+	err := s.db.Model(&storeableProcess{}).Count(&count).Error
+
+	if err != nil {
+		log.Fatal(err)
```
Comment (sfqi) [inline pkg/postalReturn/store.go:221]: Maybe return the error instead of fataling?

File: pkg/postalReturn/parser.go [hunk 4]
```diff
@@ -4,60 +4,36 @@ import (
...
-}
-
-type importer struct {
-	importInformation ImportInformation
-}
+	csvReader, err := readCSV(convertedFile)
 
-func (i *importer) readFile() *os.File {
+	if err != nil {
+		return []Process{}, []Skipped{}, errors.Wrap(err, "Could not read CSV")
```
Comment (sfqi) [inline pkg/postalReturn/parser.go:27]: Maybe `nil, nil, errors...`?

**Review by sfqi — APPROVED** [review]

File: cmd/main.go (hunk 1 above)
Comment (jpolack) [reply cmd/main.go:18]: Is for development first. In the next PR will be a proper way to configure the app.

File: pkg/postalReturn/store.go (hunk 2 above)
Comment (jpolack) [reply pkg/postalReturn/store.go:210]: If a simple operation like `Count` fails I really have internal issues. So the alternative would be to return an error and fatal more upwards the call chain, which is less verbose. What do you think?

File: pkg/postalReturn/parser.go (hunk 4 above)
Comment (jpolack) [reply pkg/postalReturn/parser.go:27]: I prefer to avoid null pointers. Is more failsafe to me.

File: pkg/postalReturn/store.go (hunk 2 above)
Comment (sfqi) [reply pkg/postalReturn/store.go:210]: Yeah, afaik the default approach to this stuff is to call panic/fatal only in main, and return errors everywhere else.

File: cmd/main.go (hunk 1 above)
Comment (jpolack) [reply cmd/main.go:18]: See: https://github.com/enercity/ed4-svc-postal-return/pull/3

File: pkg/postalReturn/store.go (hunk 2 above)
Comment (jpolack) [reply pkg/postalReturn/store.go:210]: Okay 👍  Will be fixed in https://github.com/enercity/ed4-svc-postal-return/pull/3 as well

Comment (jpolack) [reply pkg/postalReturn/store.go:210]: https://github.com/enercity/ed4-svc-postal-return/pull/3/commits/fb677386d4cb679b080ca0a97e26be23d899f625

**Pull request merged**